tests/cli/test_rasa_train.py

```diff
@@ -194,6 +199,75 @@ def test_train_force(run_in_simple_project_with_model: Callable[..., RunResult])
     assert len(files) == 2
 
 
+def test_train_dry_run(run_in_simple_project_with_model: Callable[..., RunResult]):
+    temp_dir = os.getcwd()
+
+    assert os.path.exists(os.path.join(temp_dir, "models"))
+    files = rasa.shared.utils.io.list_files(os.path.join(temp_dir, "models"))
+    assert len(files) == 1
+
+    output = run_in_simple_project_with_model("train", "--dry-run")
+    printed_output = set(output.outlines)
+
+    assert [s for s in printed_output if "No training required." in s]
+    assert output.ret == 0
+
+
+def test_train_dry_run_failure(
+    run_in_simple_project_with_model: Callable[..., RunResult]
+):
+    temp_dir = os.getcwd()
+
+    domain = (
+        "version: '2.0'\n"
+        "session_config:\n"
+        "  session_expiration_time: 60\n"
+        "  carry_over_slots_to_new_session: true\n"
+        "actions:\n"
+        "- utter_greet\n"
+        "- utter_cheer_up"
+    )
+
+    with open(os.path.join(temp_dir, "domain.yml"), "w") as f:
+        f.write(domain)
+
+    output = run_in_simple_project_with_model("train", "--dry-run")
+    printed_output = set(output.outlines)
+
+    assert not any([s for s in printed_output if "No training required." in s])
+    assert (
+        (
+            output.ret & CODE_CORE_NEEDS_TO_BE_RETRAINED
+            == CODE_CORE_NEEDS_TO_BE_RETRAINED
+        )
+        and (
+            output.ret & CODE_NLU_NEEDS_TO_BE_RETRAINED
+            == CODE_NLU_NEEDS_TO_BE_RETRAINED
+        )
+        and (
+            output.ret & CODE_NLG_NEEDS_TO_BE_RETRAINED
+            == CODE_NLG_NEEDS_TO_BE_RETRAINED
+        )
+        and (output.ret & CODE_FORCED_TRAINING != CODE_FORCED_TRAINING)
+    )
+
+
+def test_train_dry_run_force(
+    run_in_simple_project_with_model: Callable[..., RunResult]
+):
+    temp_dir = os.getcwd()
+
+    assert os.path.exists(os.path.join(temp_dir, "models"))
```

Comment:
this means that the test is dependent on the previous one which doesn't seem ideal

Reply:
Not really — it depends on `run_in_simple_project_with_model` which is specified in `conftest.py`